extensions/warp-blink-wrtc/src/rtp_logger.rs

```diff
@@ -0,0 +1,147 @@
+// this module is purposely not included in lib.rs. it might be needed later.
+
+use anyhow::{bail, Result};
+use once_cell::sync::Lazy;
+use std::{
+    fs::{self, create_dir_all},
+    io::{BufWriter, Write},
+    path::PathBuf,
+    sync::atomic::{AtomicBool, Ordering},
+};
+use tokio::sync::mpsc::{Receiver, Sender};
+use uuid::Uuid;
+use warp::sync::{Arc, RwLock};
+use webrtc::rtp::header::Header;
+
+pub struct LoggerInstance {
+    peer_id: String,
+    tx: Sender<RtpHeaderWrapper>,
+}
+
+pub struct RtpHeaderWrapper {
+    val: Header,
+    id: String,
+}
+
+impl LoggerInstance {
+    pub fn log(&self, header: Header) {
+        let _ = self.tx.try_send(RtpHeaderWrapper {
+            val: header,
+            id: self.peer_id.clone(),
+        });
+    }
+}
+
+struct RtpLogger {
+    tx: Sender<RtpHeaderWrapper>,
+    should_quit: Arc<AtomicBool>,
+    should_log: bool,
+}
+
+static RTP_LOGGER: Lazy<RwLock<Option<RtpLogger>>> =
```

Comment:
I think that wouldn't work in this case because every call to `init()` creates a new instance of `RtpLogger`. if `RTP_LOGGER` used a `OnceLock` instead of a `RwLock` then I believe it could only be written to once.

Reply:
You are correct that its only written once but it can be cleared through `OnceLock::take` which would allow you to set it again. However its not needed to be used right now as it was just a thought